src/MetadataManager.php

```diff
@@ -31,54 +31,45 @@ class MetadataManager
     private $lastError = null;
     private $serializer = null;
 
-    public function __construct($namespaceName = "Data", $containerName = "DefaultContainer")
+    public function __construct($namespaceName = "Data", $containerName = "DefaultContainer", Edmx $edmx = null)
```

Comment:
Make Edmx dependecy injectable to boost testability - can supply a mock version with, frinstance, isOK jammed open to false